quickevent/app/quickevent/plugins/Event/src/services/oresultsclient.cpp

```diff
@@ -0,0 +1,155 @@
+#include "oresultsclient.h"
+#include "oresultsclientwidget.h"
+
+#include "../eventplugin.h"
+
+#include <qf/qmlwidgets/framework/mainwindow.h>
+#include <qf/qmlwidgets/dialogs/dialog.h>
+
+#include <qf/core/log.h>
+#include <plugins/Runs/src/runsplugin.h>
+#include <plugins/Relays/src/relaysplugin.h>
+
+#include <QDir>
+#include <QFile>
+#include <QHttpPart>
+#include <QNetworkAccessManager>
+#include <QNetworkReply>
+#include <QSettings>
+#include <QStandardPaths>
+#include <QTextStream>
+#include <QTimer>
+
+namespace qfc = qf::core;
+namespace qfw = qf::qmlwidgets;
+namespace qfd = qf::qmlwidgets::dialogs;
+namespace qfs = qf::core::sql;
+using qf::qmlwidgets::framework::getPlugin;
+using Event::EventPlugin;
+using Relays::RelaysPlugin;
+using Runs::RunsPlugin;
+
+namespace Event {
+namespace services {
+
+const QString API_URL = "https://api.oresults.eu";
+
+OResultsClient::OResultsClient(QObject *parent)
+	: Super(OResultsClient::serviceName(), parent)
+{
+	m_networkManager = new QNetworkAccessManager(this);
+	m_exportTimer = new QTimer(this);
+	connect(m_exportTimer, &QTimer::timeout, this, &OResultsClient::onExportTimerTimeOut);
+	connect(this, &OResultsClient::statusChanged, [this](Status status) {
+		if(status == Status::Running) {
+			if(settings().exportIntervalSec() > 0) {
+				onExportTimerTimeOut();
+				m_exportTimer->start();
+			}
+		}
+		else {
+			m_exportTimer->stop();
+		}
+	});
+	connect(this, &OResultsClient::settingsChanged, this, &OResultsClient::init, Qt::QueuedConnection);
```

Comment:
The user must change settings to stop service in this implementation.

Reply:
I removed some redundant code. User can stop/start service from the `Services` widget. The running status is not dependent on the exportTimer interval. Saving settings does not interfere with running status.

Reply:
How can user start/stop service, if oresultclient does not implement 
```
 void run() override;
void stop() override;
```
?

Reply:
emmaclient does not implement them either
https://github.com/Quick-Event/quickbox/blob/83c0dc496abc90df879d2465df4bff3f83657d05/quickevent/app/quickevent/plugins/Event/src/services/emmaclient.h#L33-L34

Reply:
Yes, emmaclient should be fixed as well. The service can be stopped/run only by setting export time interval to 0, what is somewhat non-intuitive. I've missed it during emmaclient review. If I new this, I'd make stop/run functions pure virtual to push developers to implement Service interface.

Reply:
Emmaclient - Reaction on emit statusChanged is not enought ? run/stop only calls status update

Reply:
This type of run / stop was introduced by @fvacek in commit [QE: emma client iofxml3 export](https://github.com/Quick-Event/quickbox/commit/6c48e14844604844522d041c5f4425425c2c4101)

Reply:
> Yes, emmaclient should be fixed as well. The service can be stopped/run only by setting export time interval to 0, what is somewhat non-intuitive. I've missed it during emmaclient review. If I new this, I'd make stop/run functions pure virtual to push developers to implement Service interface.

I believe you might be incorrect. Try the artifact build. It worked for me fine. https://github.com/Quick-Event/quickbox/actions/runs/2910037225

Reply:
![image](https://user-images.githubusercontent.com/4949019/186884398-792960c7-0cd0-4c92-bf2b-97c31fb2487c.png)
Yes, I'm wrong. Anyway, current implementation where `start()/stop()` functions just change the service status, which is checked by some `if` on strange place `OResultsClient::onExportTimerTimeOut()`, is working but not easy readable, as you can see during this review:) Correct implementation should IMO start/stop export timer in `start()/stop()` re-implementations. This also applies to emmaclient.

Reply:
> Correct implementation should IMO start/stop export timer in start()/stop() re-implementations. 

I agree.

Reply:
Set up an issue and assign me to it, I will fix it later.

Reply:
https://github.com/Quick-Event/quickbox/issues/795